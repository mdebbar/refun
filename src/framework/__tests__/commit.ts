import { run, stateful, AppNode, UI } from '../';
import {
  RootTestNode,
  MockCommitNode,
  resetCommitValue,
} from '../test-helpers';

const app = (children: UI) => children;
const committer = stateful('committer', (node: AppNode, children: UI) => {
  if (node.commitNode == null) {
    node.commitNode = new MockCommitNode();
  }
  return children;
});

test('simple commit', () => {
  resetCommitValue();
  const rootCommitNode = new RootTestNode();
  run(app(committer(null)), rootCommitNode);
  expect(rootCommitNode.lastCommit).toMatchSnapshot();
});
