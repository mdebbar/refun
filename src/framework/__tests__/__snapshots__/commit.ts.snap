// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple commit 1`] = `
TestCommit {
  "$applied": 3,
  "$started": 0,
  "children": Array [
    TestCommit {
      "$applied": 2,
      "$started": 1,
      "children": Array [],
      "node": MockCommitNode {
        "lastCommit": [Circular],
      },
    },
  ],
  "node": RootTestNode {
    "lastCommit": [Circular],
  },
}
`;
